livekit-agents/livekit/agents/tokenize/_basic_word.py

```diff
@@ -1,22 +1,35 @@
 import re
 
 
-def split_words(text: str, ignore_punctuation: bool = True) -> list[str]:
+def split_words(
+    text: str, ignore_punctuation: bool = True
+) -> list[tuple[str, int, int]]:
+    """
+    Split the text into words.
+    Returns a list of words with their start and end indices of the original text.
+    """
     # fmt: off
-    punctuations = [".", ",", "!", "?", ";", ":", "'", '"', "(", ")", "[", "]", "{", "}", "<", ">",
-                    "—"]
+    punctuations = ['!', '"', '#', '$', '%', '&', "'", '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>',
```

Comment:
maybe overkill but for completeness' sake maybe we should take a slice of the unicode punctuations section?

instead of checking for each of the characters we just take its code and check if it's `0x2000 <-> 0x206F`
would be shorter too

Reply:
not sure how I can make this work with str.maketrans, not a big deal, let's keep stuff explicit :)